File: smart-contract-tutorial/examples/template.py
from boa.blockchain.vm.Ont.Runtime import Log, Notify
from boa.blockchain.vm.System.ExecutionEngine import GetScriptContainer, GetExecutingScriptHash
from boa.blockchain.vm.Ont.Transaction import *
from boa.blockchain.vm.Ont.Blockchain import GetHeight, GetHeader
from boa.blockchain.vm.Ont.Action import RegisterAction
from boa.blockchain.vm.Ont.Runtime import GetTrigger, CheckWitness
from boa.blockchain.vm.Ont.TriggerType import Application, Verification
from boa.blockchain.vm.Ont.Output import GetScriptHash, GetValue, GetAssetId
from boa.blockchain.vm.Ont.Storage import GetContext, Get, Put, Delete
from boa.blockchain.vm.Ont.Header import GetTimestamp, GetNextConsensus

def Main(operation, args):
    trigger = GetTrigger()

    if trigger == Application():
        if operation == 'add':
            a = args[0]
            b = args[1]
            return Add(a, b)

        if operation == 'compare':
            a = args[0]
            b = args[1]
            return Compare(a, b)

    return False


def Add(a, b):
    return a + b


def Compare(a, b):
    if a > b:
        return a

    return b

File: smart-contract-tutorial/examples/template.cs
using Neo.SmartContract.Framework;
using Neo.SmartContract.Framework.Services.Neo;
using Neo.SmartContract.Framework.Services.System;
using System;
using System.ComponentModel;
using System.Numerics;

namespace Neo.SmartContract
{
    public class contract1 : Framework.SmartContract
    {
        public static Object Main(string operation, params object[] args)
        {
            if (Runtime.Trigger == TriggerType.Application)
            {
                if (operation == "add")
                {
                    if (args.Length != 2) return false;
                    int a = (int)args[0];
                    int b = (int)args[1];
                    return Add(a, b);
                }
                if (operation == "compare")
                {
                    if (args.Length != 2) return false;
                    int a = (int)args[0];
                    int b = (int)args[1];
                    return Compare(a, b);
                }
            }
            return false;
        }

        public static int Add(int a, int b)
        {
            return a + b;
        }

        public static int Compare(int a, int b)
        {
            if (a > b)
            {
                return a;
            }
            return b;
        }
    }
}